### PR TITLE
First try, Big PR

### PR DESCRIPTION
**Pull request by suika93**
MANY CHANGES!1

Well this PR got bigger than I wanted it to be.
Key points:
- Implement the creation of Items and Recipes through CreateItem and CreateRecipe service objects.
- Implement Create action to both controllers: Items and Recipes.
- Remove gem Spring which was causing problems with syncing files. 

There is a lot of room for improvement, but this is a good start. :D

### CODE REVIEW

File: app/helpers/recipes_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module RecipesHelper
```
Comment (ivelten) [inline app/helpers/recipes_helper.rb]: Why is this module here? Should not we delete it if it does not have any members?

File: app/models/recipe.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Recipe < ApplicationRecord
   belongs_to :recipe_type
-  has_many :item, through: :recipe_item
+  has_many :recipe_ingredients, dependent: :destroy
+
+  validates :recipe_ingredients, length: { minimum: 2 }
```
Comment (ivelten) [inline app/models/recipe.rb]: Should not we add a maximum item too? At the moment, recipes can have a maximum of 5 items, right? We could just change this in the future if needed.

File: app/helpers/recipes_helper.rb (hunk 1 above)
Comment (suika93) [reply app/helpers/recipes_helper.rb]: It was generated with the controller. I can delete it too. We are not going to use it.

File: app/models/recipe.rb (hunk 2 above)
Comment (suika93) [reply app/models/recipe.rb]: Sure!

File: app/helpers/recipes_helper.rb (hunk 1 above)
Comment (ivelten) [reply app/helpers/recipes_helper.rb]: Oh, nevermind then. If it is auto-generated, just let it be.

**Pull request merged**